src/Http/Routing/src/Matching/DataSourceDependentMatcher.cs

```diff
@@ -41,13 +41,25 @@ public override Task MatchAsync(HttpContext httpContext)
         private Matcher CreateMatcher(IReadOnlyList<Endpoint> endpoints)
         {
             var builder = _matcherBuilderFactory();
+            var seenEndpointNames = new List<string>();
```

Comment:
```suggestion
            var seenEndpointNames = new HashSet<string>();
```

Reply:
Case-insensitive?

Reply:
Looks like they are case sensitive. The other checks don't ignore casing:

https://github.com/dotnet/aspnetcore/blob/e23fd047b4baf3480ce93d7643a897732f960557/src/Http/Routing/src/EndpointNameAddressScheme.cs#L53-L59